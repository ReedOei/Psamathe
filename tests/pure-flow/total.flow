;

3 --> var x : ! nat
[x] --> var nums : ! list ! nat
5 --> x
[x] --> nums
6 --> x
[x] --> nums

total(nums) --> var len : ! nat

