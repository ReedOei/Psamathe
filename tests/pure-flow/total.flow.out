<generatedTop>
  <k>
    .
  </k>
  <functions>
    .Map
  </functions>
  <freshCounter>
    10
  </freshCounter>
  <lookup>
    len |-> loc ( 8 )
    nums |-> loc ( 2 )
    x |-> loc ( 0 )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( nat , .List )
    loc ( 1 ) |-> resource ( nat , .List )
    loc ( 2 ) |-> resource ( list ! nat , ListItem ( ListItem ( 3 ) )
    ListItem ( ListItem ( 5 ) )
    ListItem ( ListItem ( 6 ) ) )
    loc ( 3 ) |-> resource ( list ! nat , .List )
    loc ( 4 ) |-> resource ( nat , .List )
    loc ( 5 ) |-> resource ( list ! nat , .List )
    loc ( 6 ) |-> resource ( nat , .List )
    loc ( 7 ) |-> resource ( list ! nat , .List )
    loc ( 8 ) |-> resource ( nat , ListItem ( 3 ) )
    loc ( 9 ) |-> resource ( nat , .List )
  </storage>
  <catchStack>
    .List
  </catchStack>
</generatedTop>
